api_volontaria/apps/position/models.py

```diff
@@ -0,0 +1,184 @@
+        verbose_name_plural = _("Positions")
+
+    name = models.CharField(
+        verbose_name=_("Name"),
+        max_length=100,
+    )
+
+    description = models.TextField(
+        verbose_name="Description",
+    )
+
+    hourly_wage = models.FloatField(
+        verbose_name=_("Hourly wage"),
+    )
+    
+    weekly_hours = models.FloatField(
+        verbose_name=_("Weekly hours"),
+    )
+    
+    minimum_duration_commitment = models.FloatField(
+        verbose_name=_("Minimum duration commitment"),
+    )
+
+    is_remote_job = models.BooleanField(
+        verbose_name=_("Is remote")
+    )
+    
+    is_posted = models.BooleanField(
+        verbose_name=_("Is posted")
+    )
+
+    def __str__(self):
+        return self.name
+
+    @staticmethod
+    def has_create_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def has_destroy_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def has_update_permission(request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @authenticated_users
+    def has_object_destroy_permission(self, request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+    @authenticated_users
+    def has_object_update_permission(self, request):
+        if request.user.is_staff:
+            return True
+        else:
+            return False
+
+
+class Application(models.Model):
+    """
+    This class represents an application made by volunteer for a given position.
+    """
+
+    APPLICATION_UNDER_EXAMINATION = 'UNDER_EXAMINATION'
+    APPLICATION_ACCEPTED = 'ACCEPTED'
+    APPLICATION_DECLINED = 'DECLINED'
+
+    APPLICATION_CHOICES = (
+        (APPLICATION_UNDER_EXAMINATION, _('Under examination')),
+        (APPLICATION_ACCEPTED, _('Accepted')),
+        (APPLICATION_DECLINED, _('Declined')),
+    )
+
+    class Meta:
+        verbose_name = _("Application")
+        verbose_name_plural = _("Applications")
+
+    position = models.ForeignKey(
+        Position,
+        verbose_name=_("Position"),
+        blank=False,
+        on_delete=models.CASCADE,
+    )
+    
+    applied_on = models.DateTimeField(
+        verbose_name=_("Applied on"),
+        auto_now_add=True,
+    )
+    
+    motivation = models.TextField(
+        verbose_name="Motivation",
+    )
+
+    application_status = models.CharField(
+        verbose_name=_("Application status"),
+        max_length=100,
+        choices=APPLICATION_CHOICES,
+        default=APPLICATION_UNDER_EXAMINATION
+    )
+
+    user = models.ForeignKey(
+        User,
+        related_name='applications',
+        on_delete=models.CASCADE,
+    )
+
+    def __str__(self):
+        return self.application.name
+        # return self.position.name
```

Comment:
You should remove the comment. By the way, are you sure this return is valid since `self.application` doest not seem to exist ?

Reply:
Comment removed.

def __str__ edit as follows:

    def __str__(self):
        return self.user